### ISSUE TITLE
An error occurred while Capybara/RSpec/PredicateMatcher cop was inspecting

### ISSUE THREAD

**Issue by ruban-thilak** [rubocop/rubocop#13982]
## Expected behavior

Capybara/RSpec/PredicateMatcher should not throw error while inspecting.

## Actual behavior

Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

## Steps to reproduce the problem

```
gem install rubocop -v '1.74.0'
gem install rubocop-rspec -v '2.31.0'
gem install rubocop-rails -v '2.30.1'
gem install rubocop-faker -v '1.3.0'
gem install rubocop-factory_bot -v '2.27.1'
gem install rubocop-performance -v '1.24.0'
rubocop -d --format progress
```

## RuboCop version

`1.74.0`

```
undefined method `to_sym' for nil
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/mixin/configurable_enforced_style.rb:81:in `style'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-capybara-2.22.1/lib/rubocop/cop/capybara/rspec/predicate_matcher.rb:253:in `on_send'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:143:in `public_send'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:143:in `block (2 levels) in trigger_restricted_cops'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:142:in `block in trigger_restricted_cops'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:141:in `each'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:141:in `trigger_restricted_cops'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-ast-1.38.1/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/team.rb:174:in `investigate_partial'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cop/team.rb:108:in `investigate'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:348:in `block in inspect_file'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:347:in `each'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:347:in `flat_map'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:347:in `inspect_file'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:290:in `block in do_inspection_loop'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:324:in `block in iterate_until_no_changes'
<internal:kernel>:187:in `loop'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:317:in `iterate_until_no_changes'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:286:in `do_inspection_loop'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:167:in `block in file_offenses'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:192:in `file_offense_cache'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:166:in `file_offenses'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:154:in `process_file'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:134:in `each'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:134:in `reduce'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:134:in `each_inspected_file'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:120:in `inspect_files'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/runner.rb:73:in `run'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli.rb:122:in `run_command'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli.rb:129:in `execute_runners'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli.rb:51:in `block in run'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli.rb:81:in `profile_if_needed'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/lib/rubocop/cli.rb:43:in `run'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-1.74.0/exe/rubocop:19:in `<top (required)>'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/bin/rubocop:25:in `load'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/bin/rubocop:25:in `<main>'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/bin/ruby_executable_hooks:22:in `eval'
/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/bin/ruby_executable_hooks:22:in `<main>'
An error occurred while Capybara/RSpec/PredicateMatcher cop was inspecting
```

**Comment by Earlopain** [comment]
Hi, I can't reproduce this. This looks like https://github.com/rubocop/rubocop/issues/13867 which is fixed in recent rubocop versions.

Can you provide exact steps to reproduce via gemfile/lockfile/rubocop config file etc.? The error occurs in rubocop-capybara which you don't mention in your report. Thanks!

**Comment by koic** [comment]
@ruban-thilak There may be an issue with the rubocop-factory_bot configuration. Can you take a look at it with reference to the following setup?  
https://github.com/rubocop/rubocop-factory_bot/issues/146#issuecomment-2702855960

As a side note, rubocop-factory_bot has supported plugins since version 2.27, and rubocop-capybara since version 2.22.

**Comment by ruban-thilak** [comment]
Hi, Here is my Gemfile.lock and rubocop config. I have already configured rubocop-factory_bot in the plugins section. but still facing the above mentioned error.

Gemfile.lock
```ruby
   rubocop (1.74.0)
      json (~> 2.3)
      language_server-protocol (~> 3.17.0.2)
      lint_roller (~> 1.1.0)
      parallel (~> 1.10)
      parser (>= 3.3.0.2)
      rainbow (>= 2.2.2, < 4.0)
      regexp_parser (>= 2.9.3, < 3.0)
      rubocop-ast (>= 1.38.0, < 2.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (>= 2.4.0, < 4.0)
    rubocop-ast (1.38.1)
      parser (>= 3.3.1.0)
    rubocop-capybara (2.22.1)
      lint_roller (~> 1.1)
      rubocop (~> 1.72, >= 1.72.1)
    rubocop-factory_bot (2.27.1)
      lint_roller (~> 1.1)
      rubocop (~> 1.72, >= 1.72.1)
    rubocop-faker (1.3.0)
      faker (>= 2.12.0)
      lint_roller (~> 1.1)
      rubocop (>= 1.72.1)
    rubocop-performance (1.24.0)
      lint_roller (~> 1.1)
      rubocop (>= 1.72.1, < 2.0)
      rubocop-ast (>= 1.38.0, < 2.0)
    rubocop-rails (2.30.1)
      activesupport (>= 4.2.0)
      lint_roller (~> 1.1)
      rack (>= 1.1)
      rubocop (>= 1.72.1, < 2.0)
      rubocop-ast (>= 1.38.0, < 2.0)
    rubocop-rspec (2.31.0)
      rubocop (~> 1.40)
      rubocop-capybara (~> 2.17)
      rubocop-factory_bot (~> 2.22)
      rubocop-rspec_rails (~> 2.28)
    rubocop-rspec_rails (2.29.1)
      rubocop (~> 1.61)
```

rubocop.yml
```yaml
# default configs

require:
  - rubocop-rspec
  - rubocop-rspec_rails

plugins:
  - rubocop-performance
  - rubocop-rails
  - rubocop-faker
  - rubocop-factory_bot

AllCops:
  NewCops: enable
  Exclude:
    - 'bin/**/*'
    - 'db/**/*'
    - 'vendor/**/*'
    - '.irbrc'
```

**Comment by koic** [comment]
> `/Users/rthilak/.rvm/gems/ruby-3.3.7@ams/gems/rubocop-capybara-2.22.1/lib/rubocop/cop/capybara/rspec/predicate_matcher.rb:253:in `on_send'`

Looking at the backtrace in the subject, it seems that the configuration for `rubocop-capybara` is not found. How about adding `rubocop-capybara` to `plugins`? Ultimately, the `plugins` configuration should look like the following:

```yaml
plugins:
  - rubocop-capybara
  - rubocop-faker
  - rubocop-factory_bot
  - rubocop-performance
  - rubocop-rails
  - rubocop-rspec
  - rubocop-rspec_rails
```

**Comment by ruban-thilak** [comment]
Thanks @koic, This resolved the issue. 💯

**Comment by Earlopain** [comment]
Ah, I see now. `rubocop-rspec` at `~> 2.0` has a few cops that were extracted into different gems. It then requires `rubocop-capybara` etc. in code which used to automatically modify the config. Now they don't do that anymore, having you specify it yourself. Unfortunate...
